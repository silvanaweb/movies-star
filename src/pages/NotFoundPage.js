import React from "react";
import { Link } from "react-router-dom";

const style = {
  margin: "3rem auto"
};

const NotFoundPage = () => (
  <div className="content-container">
    <div style={style}>
      Page Not Found - <Link to="/">Go home</Link>
    </div>
  </div>
);

export { NotFoundPage };
