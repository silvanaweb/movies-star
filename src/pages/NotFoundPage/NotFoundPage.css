.NotFoundPage {
  align-items: center;
  background-color: black;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.NotFoundPageGif {
  background-image: url(https://media.giphy.com/media/uLFLLXMajtNIc/giphy.gif);
  background-position-y: center;
  background-repeat: no-repeat;
  background-size: cover;
  border-bottom: 15px solid white;
  border-top: 15px solid white;
  height: 300px;
  margin-top: 2rem;
  width: 90%;
}

@media (min-width: 600px) {
  .NotFoundPageGif {
    background-size: auto;
    height: 412px;
    width: 479px;
  }
}
