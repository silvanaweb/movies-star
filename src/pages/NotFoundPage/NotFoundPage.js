import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import "./NotFoundPage.css";

const testStyle = { color: "white", fontSize: "2.8rem" };

const animateStyle = show => ({
  transition: "opacity 300ms",
  opacity: show ? 1 : 0,
  ...testStyle
});

const NotFoundPage = () => {
  const [showText, setShowText] = useState(false);
  useEffect(() => {
    setTimeout(() => {
      setShowText(true);
    }, 2000);
  }, []);

  return (
    <div className="NotFoundPage">
      <div style={animateStyle(showText)}>
        <span>Got lost, huh?</span>{" "}
        <Link style={testStyle} to="/">
          Find your way home =>
        </Link>
      </div>
      <div className="NotFoundPageGif" />
    </div>
  );
};

export { NotFoundPage };
