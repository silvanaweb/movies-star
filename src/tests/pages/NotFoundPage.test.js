import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import { NotFoundPage } from "../../pages/NotFoundPage";

let wrapper, history;

beforeEach(() => {
  history = { push: jest.fn() };
  wrapper = shallow(<NotFoundPage history={history} />);
});

test("should render NotFoundPage correctly", () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should handle onGoToHome correctly", () => {
  wrapper.find("button").simulate("click");
  expect(history.push).toHaveBeenLastCalledWith("/");
});
