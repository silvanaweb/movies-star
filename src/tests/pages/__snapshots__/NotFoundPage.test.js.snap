// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotFoundPage correctly 1`] = `
<div
  className="content-container"
>
  <div
    style={
      Object {
        "margin": "3rem auto",
      }
    }
  >
    Page Not Found -
     
    <button
      className="Button--link"
      onClick={[Function]}
    >
      Go home
    </button>
  </div>
</div>
`;
